test/unittests/evm_memory_test.cpp

```diff
@@ -0,0 +1,246 @@
+    {OP_EXTCODECOPY, 1, 3},
+    {OP_RETURNDATACOPY, 0, 2},
+    {OP_LOG0, 0, 1},
+    {OP_LOG1, 0, 1},
+    {OP_LOG2, 0, 1},
+    {OP_LOG3, 0, 1},
+    {OP_LOG4, 0, 1},
+    {OP_RETURN, 0, 1},
+    {OP_REVERT, 0, 1},
+    {OP_CALL, 3, 4},
+    {OP_CALL, 5, 6},
+    {OP_CALLCODE, 3, 4},
+    {OP_CALLCODE, 5, 6},
+    {OP_DELEGATECALL, 2, 3},
+    {OP_DELEGATECALL, 4, 5},
+    {OP_STATICCALL, 2, 3},
+    {OP_STATICCALL, 4, 5},
+    {OP_CREATE, 1, 2},
+    {OP_CREATE2, 1, 2},
+};
+
+struct
+{
+    uint256 index;
+    uint256 size;
+} memory_access_test_cases[] = {
+    {0, 0x100000000},
+    {0, 0x10000000000000000_u256},
+    {0, 0x100000000000000000000000000000000_u256},
+    {0, 0x1000000000000000000000000000000000000000000000000_u256},
+    {0, 0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff_u256},
+    {0x80000000, 0x80000000},
+    {0xffffffe0, 32},
+    {0xffffffff, 1},
+    {0x100000000, 0},
+    {0x100000000, 1},
+    {0x100000000, 0x100000000},
+    {0xffffffffffffffc1, 32},
+    {0xffffffffffffffc0, 32},
+    {0xffffffffffffffdf, 32},
+    {0xffffffffffffffe0, 32},
+    {0xffffffffffffffff, 1},
+    {0x80000000000000000000000000000000_u256, 1},
+    {0x800000000000000000000000000000000000000000000000_u256, 1},
+    {0x8000000000000000000000000000000000000000000000000000000000000000_u256, 1},
+    {0x8000000000000000800000000000000080000000000000000000000000000000_u256, 1},
+    {0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff_u256, 1},
+    {0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff_u256,
+        0xffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff_u256},
+};
+
+TEST_P(evm, memory_access)
+{
+    // This test checks if instructions accessing memory properly respond with out-of-gas
+    // error for combinations of memory offset and memory size arguments.
+    rev = EVMC_CONSTANTINOPLE;
+
+    for (const auto& p : memory_access_test_cases)
+    {
+        for (auto& t : memory_access_opcodes)
+        {
+            const auto num_args = int{instr::traits[t.opcode].stack_height_required};
+            auto h = std::max(num_args, t.memory_size_arg + 1);
+            auto code = bytecode{};
+
+            if (t.memory_size_arg >= 0)
+            {
+                while (--h != t.memory_size_arg)
+                    code += push(0);
+
+                code += push(p.size);
+            }
+            else if (p.index == 0 || p.size == 0)
+                continue;  // Skip opcodes not having SIZE argument.
+
+            while (--h != t.memory_index_arg)
+                code += push(0);
+
+            code += push(p.index);
+
+            while (h-- != 0)
+                code += push(0);
+
+            code += bytecode{t.opcode};
+
+            const auto gas = 8796294610952;
+            execute(gas, code);
+
+            auto case_descr_str = std::ostringstream{};
+            case_descr_str << "offset = 0x" << std::hex << to_string(p.index) << " size = 0x"
```

Comment:
Does `to_string` output it in hexadecimal? then `std::hex` is not needed?

Reply:
Converted to proper hex.